src/HttpClientPool/RoundRobinClientPool.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Http\Client\Common\HttpClientPool;
+
+use Http\Client\Common\Exception\HttpClientNotFoundException;
+use Http\Client\Common\HttpClientPool;
+
+/**
+ * RoundRobinClientPool will choose the next client in the pool.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class RoundRobinClientPool extends HttpClientPool
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function chooseHttpClient()
+    {
+        $last = current($this->clientPool);
```

Comment:
both are only within the same php process. when using a client in web requests, there hopefully is only one request to some other service, not several, and in that case this only makes sense if we randomize